### PR TITLE
Update to SNAPSHOT #7141639

### PR DESCRIPTION
**Pull request by chrisbanes**

### CODE REVIEW

**Review by JoseAlcerreca — APPROVED** [review]

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt [hunk 1]
```diff
@@ -154,39 +156,41 @@ class ImeNestedScrollConnection(
...
 
         if (imeAnimController.isInsetAnimationInProgress()) {
             // If we have an IME animation in progress, from the user scrolling, we can
             // animate to the end state using the velocity
-            imeAnimController.animateToFinish(available.y) { remainingVelocity ->
-                onFinished(Velocity(x = 0f, y = remainingVelocity))
+            return suspendCancellableCoroutine { cont ->
+                imeAnimController.animateToFinish(available.y) { remainingVelocity ->
+                    cont.resume(Velocity(x = 0f, y = remainingVelocity)) {
+                        imeAnimController.cancel()
```
Comment (nickbutcher) [inline insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt:175]: Should this `invokeOnCancellation` as well?

Comment (nickbutcher) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt:175]: If not, use `suspendCoroutine`?

Comment (manuelvicnt) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt:175]: That's what the resume block does https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-cancellable-continuation/resume.html

But maybe if you just use `.invokeOnCancellation`, you can avoid the experimental annotation

Comment (chrisbanes) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/ImeNestedScrollConnection.kt]: Not quite. The `onCancellation` block on `resume()` is used for when the resume itself fails (due the outer coroutines being cancelled, etc). `invokeOnCancellation` is used for the normal cancelled state whilst suspended event.

Anyway, added `invokeOnCancellation`

**Review by manuelvicnt — APPROVED** [review]

**Review by nickbutcher — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**